dlt/common/normalizers/json/relational.py

```diff
@@ -423,10 +423,21 @@ def _normalize_prop(
         )
 
     @staticmethod
-    def _get_table_nesting_level(schema: Schema, table_name: str) -> Optional[int]:
+    def _get_table_nesting_level(
+        schema: Schema, table_name: str, parent_path: Tuple[str, ...]
+    ) -> Optional[int]:
+        """gets table nesting level, will inherit from parent if not set"""
+
+        # try go get table directly
         table = schema.tables.get(table_name)
+
+        # if table is not found, try to get it from root path
+        if not table and parent_path:
+            table = schema.tables.get(parent_path[0])
```

Comment:
should we check intermediary tables here too? so if parent path is ["one", "two"] should we also check table "one__two" or is it always correct to use the top table? I'm not sure there is a case where there would be settings in a intermediary table.

Reply:
we need only parent table. the overall fix is good but maybe we could optimize this block a little? it is always top level table that has this setting  so maybe we should pass root table as table_name here? then no changes are needed